src/test/regress/sql/multi_task_assignment_policy.sql

```diff
@@ -18,17 +18,16 @@ DECLARE
        shardOfTheTask text;
 begin
   for r in execute qry loop
-       IF r LIKE '%port%' THEN
-       portOfTheTask = substring(r, '([0-9]{1,10})');
+    IF r LIKE '%port%' THEN
+      portOfTheTask = substring(r, '([0-9]{1,10})');
     END IF;
 
     IF r LIKE '%' || table_name || '%' THEN
-       shardOfTheTask = substring(r, '([0-9]{1,10})');
-       return QUERY SELECT  shardOfTheTask || '@' || portOfTheTask ;
```

Comment:
On some queries the regex for `shardOfTheTask` captures a wrong number when there is a line containing the table name, but no shard ids. For example `([0-9]{1,10})` captures `0` in the following query plan.

I moved the return statement after the end of the loop, so that all regex captures will overwrite the values of variables `portOfTheTask` and `shardOfTheTask` and the latest ones will be returned by the function.

```
                                                         QUERY PLAN
-----------------------------------------------------------------------------------------------------------------------------
 Custom Scan (Citus Router)  (cost=0.00..0.00 rows=0 width=0)
   ->  Distributed Subplan 1_1
         ->  Seq Scan on task_assignment_nonreplicated_hash  (cost=0.00..38.25 rows=11 width=8)
               Filter: (test_id = 3)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=9702 dbname=postgres
         ->  Merge Left Join  (cost=239.61..435.86 rows=12750 width=12)
               Merge Cond: (intermediate_result.ref_id = ref.test_id)
               ->  Sort  (cost=59.83..62.33 rows=1000 width=8)
                     Sort Key: intermediate_result.ref_id
                     ->  Function Scan on read_intermediate_result intermediate_result  (cost=0.00..10.00 rows=1000 width=8)
               ->  Sort  (cost=179.78..186.16 rows=2550 width=4)
                     Sort Key: ref.test_id
                     ->  Seq Scan on task_assignment_reference_table_102042 ref  (cost=0.00..35.50 rows=2550 width=4)
(16 rows)
```